api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -742,6 +743,45 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
     return ResponseEntity.ok(response);
   }
 
+  @Override
+  public ResponseEntity<EmptyResponse> renameNotebook(String workspace, String workspaceName, NotebookRename rename) {
+    String fromBucket = fireCloudService.getWorkspace(workspace, workspaceName)
+      .getWorkspace()
+      .getBucketName();
+    String origPath = "notebooks/" + rename.getName();
+    String newName = rename.getNewName();
+    if (!NOTEBOOK_PATTERN.matcher(newName).matches()) {
+      newName = newName + ".ipynb";
+    }
+    String newPath = "notebooks/" + newName;
+    BlobId blobId = BlobId.of(fromBucket, origPath);
+    cloudStorageService.copyBlob(blobId, BlobId.of(fromBucket, newPath));
+    cloudStorageService.deleteBlob(blobId);
+    return ResponseEntity.ok(new EmptyResponse());
+  }
+
+  @Override
+  public ResponseEntity<EmptyResponse> cloneNotebook(String workspace, String workspaceName, String notebookName) {
+    String fromBucket = fireCloudService.getWorkspace(workspace, workspaceName)
+      .getWorkspace()
+      .getBucketName();
+    String origPath = "notebooks/" + notebookName;
```

Comment:
NIT:Will it make sense to have "notebooks" as  a file const variable since its used in multiple places?